blockstore/apps/search/fieldsets.py

```diff
@@ -0,0 +1,119 @@
+"""
+FieldSets for search app.
+"""
+from elasticsearch_dsl import (
+    Keyword,
+    InnerDoc as ESInnerDoc,
+    Integer,
+    Text,
+)
+
+from rest_framework import serializers
+from .serializers import FieldSetSerializer
+
+
+class InnerDoc(ESInnerDoc):
+    pass
+
+
+class FieldSet:
+    """
+    A FieldSet describes a collection of ElasticSearch fields which can be
+    dynamically associated with an Index to add support for indexing metadata
+    from different sources related to entities in that Index.
+    """
+
+    name = None  # A string used to namespace the fields.
+    Document = None  # An ElasticSearch InnerDoc.
+    Serializer = None  # A DRF Serializer which can be use to serialize and deserialize the FieldSet.
+
+
+class SummaryFieldSet(FieldSet):
+    """
+    FieldSet for summary data of the entity.
+    """
+
+    name = 'summary'
+
+    class Document(InnerDoc):
+
+        title = Text()
+        description = Text()
+        image = Text(index=False)
+
+    class Serializer(FieldSetSerializer):
+
+        title = serializers.CharField(required=False)
+        description = serializers.CharField(required=False)
+        image = serializers.URLField(required=False)
+
+
+class OwnernershipFieldSet(FieldSet):
+    """
+    FieldSet for ownership data of the entity.
+    """
+
+    name = 'ownership'
+
+    class Document(InnerDoc):
+
+        org_id = Keyword()
+
+    class Serializer(FieldSetSerializer):
+
+        org_id = serializers.CharField(required=False)
+
+
+class AuthorshipFieldSet(FieldSet):
+    """
+    FieldSet for authorship data of the entity.
+    """
+
+    name = 'authorship'
+
+    class Document(InnerDoc):
+
+        author_ids = Keyword(multi=True)
```

Comment:
For the short term we can put the names in here.

For the long term, there are a few interesting questions here. For objects (author or org in this case) which can be associated with many entities in the index, do we just want to store stable identifiers or other metadata as well with each entity? If we do, in case the metadata (author name or org name) is edited, it would require re-indexing all the entities with which that metadata was stored.

The other option is that since author and org metadata is going to be stored in Blockstore, we can fill in the values for the results which are being returned in the query.

One other consideration is the performance implications of the exact data structure we use to store the metadata. `nested` fields results in each nested object indexed as a separate document while object arrays do not; but have querying limitations.

https://www.elastic.co/guide/en/elasticsearch/reference/current/nested.html

Reply:
Since we do not have a model for storing authors and owners (any related metadata), we can just put the whole name here for now. See tests for example.

Reply:
Hmm. I have avoided adding any object arrays for now since they have some [peculiar behaviors](https://www.elastic.co/guide/en/elasticsearch/reference/current/nested.html#_how_arrays_of_objects_are_flattened) that I need to understand better. So lets go with storing the id for now.

Reply:
We will be transforming the ES responses in LX service anyway, to for example map the full tag paths to their labels etc so this can just be another transform which adds data.